src/tooltip.tsx

```diff
@@ -135,10 +142,64 @@ const Tooltip: React.FC<Props> = ({
         height: 0,
     });
 
-    const getPosition = (position: Position = defaultPositionDesktop) =>
-        isTabletOrSmaller && (position === 'left' || position === 'right') ? defaultPositionMobile : position;
+    const [containerPosition, setContainerPosition] = React.useState({});
+
+    const getPosition = (position: Position | undefined) =>
+        isTabletOrSmaller ? position || defaultPositionMobile : position || defaultPositionDesktop;
+
+    const noRightSpace = (position: Position) => {
+        if (!tooltipRef.current) return position;
+        const tooltipBoundingClientRect = tooltipRef.current.getBoundingClientRect();
```

Comment:
`getBoundingClientRect` is an expensive operation, you could call it just once, and with the result then calculate if there is enough right/left/top/bottom space